languages/python/TreeSitter/Python.hs

```diff
@@ -4,14 +4,5 @@ module TreeSitter.Python
 
 import Foreign.Ptr
 import TreeSitter.Language
-import CodeGen.Deserialize
-import CodeGen.GenerateSyntax
-import Control.Monad.IO.Class
```

Comment:
No need to modify each of these subprojects: we can depend on them in the `codegen` subproject.

Reply:
I think this would result in worse usability than @aymannadeem’s approach here. Ideally, a consumer of `tree-sitter-python` should get the datatypes and functions to work with strongly-typed ASTs right out of the box.

Generating the sources fresh for each consumer isn’t really necessary, since this isn’t like e.g. protobuf codegen for some .proto definitions; these datatypes don’t (and shouldn’t) e.g. comprise a wire protocol.

What do you think?

Reply:
I don't have strong feels either way, I did this with the intent of originally having `input` defined here, and having the splicing happen in `/languages/python/TreeSitter/CodeGen.hs` on my branch.